src/Uno.UI/UI/Xaml/UIElement.Pointers.macOS.cs

```diff
@@ -0,0 +1,192 @@
+			AddTrackingArea(_trackingArea);
+		}
+
+		public override void MouseDown(NSEvent evt)
+		{
+			try
+			{
+				// evt.AllTouches raises a invalid selector exception
+				var args = new PointerRoutedEventArgs(null, evt, this);
+				 
+				var pointerEventIsHandledInManaged = OnNativePointerDown(args);
+
+				if (!pointerEventIsHandledInManaged)
+				{
+					// Bubble up the event natively
+					base.MouseDown(evt);
+				}
+			}
+			catch (Exception e)
+			{
+				Application.Current.RaiseRecoverableUnhandledException(e);
+			}
+		}
+
+		public override void MouseMoved(NSEvent evt)
+		{
+			try
+			{
+				// evt.AllTouches raises a invalid selector exception
+				var args = new PointerRoutedEventArgs(null, evt, this);				
+
+				var pointerEventIsHandledInManaged = OnNativePointerMove(args);
+
+				if (!pointerEventIsHandledInManaged)
+				{
+					// Bubble up the event natively
+					base.MouseMoved(evt);
+				}
+			}
+			catch (Exception e)
+			{
+				Application.Current.RaiseRecoverableUnhandledException(e);
+			}
+		}
+
+		public override void MouseEntered(NSEvent evt)
+		{
+			try
+			{
+				var args = new PointerRoutedEventArgs(null, evt, this);
+				
+				var pointerEventIsHandledInManaged = OnNativePointerEnter(args);
+
+				if (!pointerEventIsHandledInManaged)
+				{
+					// Bubble up the event natively
+					base.MouseEntered(evt);
+				}
+			}
+			catch (Exception e)
+			{
+				Application.Current.RaiseRecoverableUnhandledException(e);
+			}
+		}
+
+		public override void MouseExited(NSEvent evt)
+		{
+			try
+			{
+				// evt.AllTouches raises a invalid selector exception
+				var args = new PointerRoutedEventArgs(null, evt, this);				 
+
+				var pointerEventIsHandledInManaged = OnNativePointerExited(args);
+
+				if (!pointerEventIsHandledInManaged)
+				{
+					// Bubble up the event natively
+					base.MouseExited(evt);
+				}
+			}
+			catch (Exception e)
+			{
+				Application.Current.RaiseRecoverableUnhandledException(e);
+			}
+		}
+
+		public override void MouseUp(NSEvent evt)
+		{
+			try
+			{
+				// evt.AllTouches raises a invalid selector exception
+				var args = new PointerRoutedEventArgs(null, evt, this);
+				var pointerEventIsHandledInManaged = false;
+
+				pointerEventIsHandledInManaged = OnNativePointerUp(args);
+
+				if (!pointerEventIsHandledInManaged)
+				{
+					// Bubble up the event natively
+					base.MouseUp(evt);
+				}
+			}
+			catch (Exception e)
+			{
+				Application.Current.RaiseRecoverableUnhandledException(e);
+			}			
+		}
+
+		public override void MouseDragged(NSEvent evt)
+		{
+			try
+			{
+				var wasPointerOver = IsPointerOver;
+				IsPointerOver = evt.IsTouchInView(this);
+
+				var pointerEventIsHandledInManaged = false;
+
+				// evt.AllTouches raises a invalid selector exception
+				var args = new PointerRoutedEventArgs(null, evt, this);
+
+				if (IsPointerCaptured || IsPointerOver)
+				{
+					pointerEventIsHandledInManaged = OnNativePointerMove(args);
```

Comment:
Here instead of validating the `IsPointerCaptured` and `IsPointerOver` you should be able to use the `OnNativePointerMoveWithOverCheck(args, evt.IsTouchInView(this))` which already does that. 

BTW here we should update the over state (i.e. enter / exit) before raising the move since the coordinates of the `args` are already reflecting this updated state.

Reply:
Will do :-)

Reply:
Is the current version what you meant? The `OnNativePointerMoveWithOverCheck` triggers the Enter/Exit events automatically in case it is necessary, so it is ok to remove the other ones?

Reply:
Yes it does, that the purpose of the [`SetOver`](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.cs#L663) 😀

Reply:
Thanks 👍👍